clippy_lints/src/default_trait_access.rs

```diff
@@ -48,6 +48,13 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for DefaultTraitAccess {
             then {
                 match qpath {
                     QPath::Resolved(..) => {
+                        if let ExprKind::Call(ref method, ref _args) = expr.node {
+                            if format!("{:?}", method).contains(" as Default>") {
```

Comment:
:laughing: great hack, but I think there could be a more structured solution. Have you tried dumping the contents of `method.node`?

Reply:
I did some poking and some learning and some reading of rustc, and hopefully this looks much better now :)